packages/memory-inspector/src/browser/editable-widget/memory-editable-table-widget.tsx

```diff
@@ -0,0 +1,324 @@
+            const { value: inMemory } = this.composeByte(address, false);
+            const oldValue = this.previousBytes && this.composeByte(address, false, this.previousBytes).value;
+            const decimal = parseInt(value, 16);
+            const octal = decimal.toString(8).padStart(this.options.byteSize / 8, '0');
+            const UTF8 = String.fromCharCode(decimal);
+            const binary = this.getPaddedBinary(decimal);
+            const toSend: EasilyMappedObject = { hex: value, octal, binary, decimal };
+            if (UTF8) {
+                toSend.UTF8 = UTF8;
+            }
+            if (inMemory !== value) {
+                toSend['Current Value'] = inMemory;
+            }
+            if (oldValue !== undefined && oldValue !== value) {
+                toSend['Previous Value'] = oldValue;
+            }
+            return toSend;
+        }
+        return undefined;
+    }
+
+    protected composeByte(
+        addressPlusArrayOffset: Long,
+        usePendingEdits: boolean,
+        dataSource: Uint8Array = this.memory.bytes,
+    ): Interfaces.ByteFromChunkData {
+        let value = '';
+
+        const offset = addressPlusArrayOffset.subtract(this.memory.address);
+        const chunksPerByte = this.options.byteSize / 8;
+        const startingChunkIndex = offset.subtract(offset.modulo(chunksPerByte));
+        const address = this.memory.address.add(startingChunkIndex.divide(chunksPerByte));
+        for (let i = 0; i < chunksPerByte; i += 1) {
+            const targetOffset = startingChunkIndex.add(i);
+            const targetChunk = this.getFromMapOrArray(targetOffset, usePendingEdits, dataSource);
+            value += targetChunk.padStart(2, '0');
+        }
+
+        return { address, value };
+    }
+
+    protected getFromMapOrArray(arrayOffset: Long, usePendingEdits: boolean, dataSource: Uint8Array = this.memory.bytes): string {
+        let value = usePendingEdits ? this.pendingMemoryEdits.get(arrayOffset.add(this.memory.address).toString()) : undefined;
+        if (value === undefined) {
+            value = dataSource[arrayOffset.toInt()]?.toString(16) ?? '';
+        }
+        return value;
+    }
+
+    protected handleClearEditClick = (): void => this.clearEdits();
+
+    protected clearEdits(address?: Long): void {
+        if (typeof address === 'number') {
+            this.pendingMemoryEdits.delete(address);
+        } else {
+            this.pendingMemoryEdits.clear();
+        }
+        this.update();
+    }
+
+    protected submitMemoryEdits = async (): Promise<void> => {
+        this.memoryEditsCompleted = new Deferred();
+        for (const edit of this.createUniqueEdits()) {
+            try {
+                await this.memoryProvider.writeMemory(edit);
+            } catch (e) {
+                console.log('Problem writing memory with arguments', edit, '\n', e);
+            }
+        }
+        this.memoryEditsCompleted.resolve();
```

Comment:
I noticed that when I submit edits there is no update on the view by default. We do resolve the edits which means that the next time the memory is changed it will update accordingly and the pending edits are cleared (`handleMemoryChange`). But this requires a specific user action. At the moment, the pending changes just remain in the UI. Should we trigger an explicit update from here or what is the expected flow after the user submits the edits?

Reply:
This is a bit tricky because the debug session doesn't emit any events when memory changes - we just get a response. When we implemented this internally, we just had the debug adapter send a `StoppedEvent` after successfully updating memory, but that was a bit of a workaround, and I didn't include it when implementing the write memory request [upstream](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/blob/d8c6ab55a03025f6facf917cca8eaed69b19fe76/src/GDBDebugSession.ts#L1373-L1408) in the CDT-GDB adapter. I think your suggestion of triggering a refresh cycle is probably the best way to handle it.

Reply:
I've modified the code here so that if a write succeeds we refresh the memory, but we don't clear any pending edits.